src/io/flutter/actions/FlutterAppAction.java

```diff
@@ -17,21 +18,22 @@
 abstract public class FlutterAppAction extends DumbAwareAction {
 
   private final ObservatoryConnector myConnector;
-  private Computable<Boolean> myIsApplicable;
+  private final Computable<Boolean> myIsApplicable;
   private FlutterApp.State myAppState;
   private boolean myIsListening = false;
-  private FlutterApp.StateListener myListener = new FlutterApp.StateListener() {
+  private final FlutterApp.StateListener myListener = new FlutterApp.StateListener() {
     @Override
     public void stateChanged(FlutterApp.State newState) {
       myAppState = newState;
       getTemplatePresentation().setEnabled(myIsApplicable.compute() && isRunning());
     }
   };
 
-  public FlutterAppAction(ObservatoryConnector connector, String text, String description, Icon icon, Computable<Boolean> isApplicable) {
+  public FlutterAppAction(ObservatoryConnector connector, String text, String description, Icon icon, Computable<Boolean> isApplicable, @NotNull String actionId) {
     super(text, description, icon);
     myConnector = connector;
     myIsApplicable = isApplicable;
+    ActionManager.getInstance().registerAction(actionId, this);
```

Comment:
This object is registering itself globally in its ctor; I assume this ctor is only called once?